test/jdk/java/foreign/TestIntrinsics.java

```diff
@@ -32,6 +32,21 @@
  *   TestIntrinsics
  */
 
+/*
+ * @test
+ * @bug 8286681
+ * @summary ShenandoahControlThread::request_gc misses the case of GCCause::_codecache_GC_threshold
+ * @enablePreview
+ * @requires os.arch=="amd64" | os.arch=="x86_64" | os.arch=="aarch64"
+ * @requires vm.gc.Shenandoah
+ * @run testng/othervm
+ *   -Djdk.internal.foreign.ProgrammableInvoker.USE_SPEC=true
+ *   --enable-native-access=ALL-UNNAMED
+ *   -Xbatch
+ *   -XX:+UseShenandoahGC
+ *   TestIntrinsics
+ */
```

Comment:
Is this needed? The parameters looks the same as the first test description so if you are testing with +ShenandoahGC then it will run already.

Reply:
Without `-XX:+UseShenandoahGC`, this bug wouldn't be exposed.

What do you mean by `if you are testing with +ShenandoahGC then it will run already`?

Reply:
I assume you are running the tests with:
   make run-tests TEST_OPTS_JAVA_OPTIONS="-XX:+UseShenandoahGC" 
in which case, all of the tests you select to run will be run with that GC. 

What you have is not wrong but wouldn't be a better to add a new test to test/hotspot/jtreg/gc rather than relying on test in test/jdk/java/foreign?

Reply:
> I assume you are running the tests with: make run-tests TEST_OPTS_JAVA_OPTIONS="-XX:+UseShenandoahGC" in which case, all of the tests you select to run will be run with that GC.

Yes, you're right.

> What you have is not wrong but wouldn't be a better to add a new test to test/hotspot/jtreg/gc rather than relying on test in test/jdk/java/foreign?

I would suggest reusing the existing test in java/foreign.
This is because this bug was first triggered after `Implementation of Foreign Function & Memory API (Preview)` integration.
And I don't want a copy-paste code duplication in a new test file.
Thanks.

Reply:
I think I agree with @AlanBateman - in the sense that this seems to go down a slippery slope where every test would need to be executed against all possible GCs. AFAIK, there are no other foreign tests doing this.

Reply:
> I think I agree with @AlanBateman - in the sense that this seems to go down a slippery slope where every test would need to be executed against all possible GCs. AFAIK, there are no other foreign tests doing this.

I think it's a waste of time to write a separate test for this bug.

I can't understand why you are against adding one more test config in the foreign test.
Yes, it caught the bug in ShenandoahGC but who knows it wouldn't find some other bugs in the foreign api in the future.
If you are still against the newly added test config, I can revert the test change.
Thanks.

Reply:
My comment seems to have never made it through. The problem with what your are doing is two-fold:
1. When running the test suite specifically to test xGC for whatever x you are now forcing a test run for Shenandoah.
2.  When x is Shenandoah then you run this test twice.

You don't need a config just to run this with Shenandoah - you run the test suite with Shenandoah.

Reply:
The test change had been reverted.
Thanks.